### ISSUE TITLE
Training doesn't work

### ISSUE THREAD

**Issue by andrew-hewitt** [wit-ai/wit#495]
**Do you want to request a *feature*, report a *bug*, or ask a *question* about wit?**
Report a possible bug

**What is the current behaviour?**
Bot refuses to obey new training. Can't stop insisting on its first impression. Impossible to convince it that it is wrong. Basically acting too Human to be useful.

**If the current behaviour is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

1 - Create new story named "set Age" with the user input as "I am 30 years old".
2 - Save and wait a little. After a short time this is flagged as "could not understand."

![actionslist](https://cloud.githubusercontent.com/assets/22877956/25033206/8d62ba08-2116-11e7-9207-3a1a8fe3013b.png)

![cantunderstand](https://cloud.githubusercontent.com/assets/22877956/25033219/a2a14f7e-2116-11e7-8414-b1b140ee7077.png)

3 - Go to understanding and train it...

3.1 - Gets it totally wrong...

![wrong](https://cloud.githubusercontent.com/assets/22877956/25033234/ceacf654-2116-11e7-9d88-f24a0b27b488.png)

3.2 - Retrain...

![retrained](https://cloud.githubusercontent.com/assets/22877956/25033247/e7405576-2116-11e7-9c7a-c8a2d94bedd3.png)

3.3 - Re-test....

![wrongagain](https://cloud.githubusercontent.com/assets/22877956/25033261/10e75262-2117-11e7-909a-6fb17480e749.png)

4 - Return to step 1.

NOTE: I can see that the intent has been correctly identified. But not the fact this is an age.
It will also, occasionally, decide "I am" means "I" is a first_name for a contact.

**What is the expected behaviour?**

After training, the bot should recognise the new behaviour and stop insisting that age is a duration, and NOT return "I" as a first name.

It is possible we are forgetting a step. Or getting the whole thing wrong. In which case I would love to know how to do it. ^^;

**Comment by l5t** [comment]
can you share your Wit app url?

**Comment by andrew-hewitt** [comment]
Actually we have developed a work around which has satisfied our means.
However, to demonstrate the above problem I created a simple test bot which does the same thing. In doing so I realised that the problem is confusion between the inbuilt wit/age_of_person and wit/duration entities. This doesn't explain the "I" being set as the contact, but presumably this is also down to confusion when creating a new story for person name.

https://wit.ai/andrew-hewitt/MrTestBot

I would think training should help the bot stop making the mistake. But it seems duration takes priority over age.

Our workaround was to stop using inbuilt wit entities in our main project. Now we create all of our own entities and everything works as expected.
